shared-bindings/countio/LinearEncoder.c

```diff
@@ -0,0 +1,141 @@
+
+#include <stdint.h>
+
+#include "lib/utils/context_manager_helpers.h"
+#include "py/objproperty.h"
+#include "py/runtime.h"
+#include "py/runtime0.h"
+#include "shared-bindings/microcontroller/Pin.h"
+#include "shared-bindings/countio/LinearEncoder.h"
+#include "shared-bindings/util.h"
+
+//| .. currentmodule:: countio
+//|
+//| :class:`LinearEncoder` -- Track the relative position of an linear encoder
```

Comment:
I don't think this should be called LinearEncoder because it's simply counting the number of edges since startup. I'd just call it `Counter`. It might be worth adding a reset() too.